presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -231,11 +231,7 @@ public void checkCanSetUser(Optional<Principal> principal, String userName)
             return;
         }
 
-        if (principal.isEmpty()) {
-            denySetUser(principal, userName);
-        }
-
-        String principalName = principal.get().getName();
+        String principalName = (principal.isPresent()) ? principal.get().getName() : "";
```

Comment:
I think we need dedicated rule for that:

```
{
  "catalogs": [
    {
      "allow": true
    }
  ],
  "principals": [
   ...
  ]`,
 unathenticated_user = "bob"
}
```

Other than that, @dain shouldn't we allow all in `checkCanSetUser` here and rely only on  `checkCanImpersonateUser`?

Reply:
Alternative approach is to simply merge: https://github.com/prestosql/presto/pull/4081